graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -382,6 +382,8 @@ def _compile_vertex_ast(schema, current_schema_type, ast,
         in_topmost_optional_block = False
 
         edge_traversal_is_optional = optional_directive is not None
+        in_optional_context = 'optional' in context and not edge_traversal_is_optional
```

Comment:
Is this correct? What if we have an optional traversal that is already inside of an optional context -- won't this end up being `False` in that case? That seems questionable.

In either case, let's cover it with a test.

Also, I'm not sure what the desired semantics of `in_optional_context` are -- is it meant to only be True when within an existing optional context, but not when opening the initial optional context? A comment + maybe a variable name tweak might be useful.